pkg/api/handlers/catalogs.go

```diff
@@ -0,0 +1,385 @@
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"github.com/obot-platform/obot/pkg/system"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type MCPCatalogHandler struct {
+	allowedDockerImageRepos []string
+}
+
+func NewCatalogHandler(allowedDockerImageRepos []string) *MCPCatalogHandler {
+	return &MCPCatalogHandler{
+		allowedDockerImageRepos: allowedDockerImageRepos,
+	}
+}
+
+// List returns all catalogs.
+func (*MCPCatalogHandler) List(req api.Context) error {
+	var list v1.MCPCatalogList
+	if err := req.List(&list); err != nil {
+		return fmt.Errorf("failed to list catalogs: %w", err)
+	}
+
+	var items []types.MCPCatalog
+	for _, item := range list.Items {
+		items = append(items, types.MCPCatalog{
+			Metadata: MetadataFrom(&item),
+			MCPCatalogManifest: types.MCPCatalogManifest{
+				DisplayName:    item.Spec.DisplayName,
+				SourceURLs:     item.Spec.SourceURLs,
+				AllowedUserIDs: item.Spec.AllowedUserIDs,
+			},
+			IsReadOnly: item.Spec.IsReadOnly,
+			LastSynced: *types.NewTime(item.Status.LastSyncTime.Time),
+		})
+	}
+
+	return req.Write(types.MCPCatalogList{
+		Items: items,
+	})
+}
+
+// Get returns a specific catalog by ID.
+func (*MCPCatalogHandler) Get(req api.Context) error {
+	var catalog v1.MCPCatalog
+	if err := req.Get(&catalog, req.PathValue("catalog_id")); err != nil {
+		return fmt.Errorf("failed to get catalog: %w", err)
+	}
+	return req.Write(types.MCPCatalog{
+		Metadata: MetadataFrom(&catalog),
+		MCPCatalogManifest: types.MCPCatalogManifest{
+			DisplayName:    catalog.Spec.DisplayName,
+			SourceURLs:     catalog.Spec.SourceURLs,
+			AllowedUserIDs: catalog.Spec.AllowedUserIDs,
+		},
+		IsReadOnly: catalog.Spec.IsReadOnly,
+		LastSynced: *types.NewTime(catalog.Status.LastSyncTime.Time),
+	})
+}
+
+// Refresh refreshes a catalog to sync its entries.
+func (h *MCPCatalogHandler) Refresh(req api.Context) error {
+	catalogName := req.PathValue("catalog_id")
+
+	var catalog v1.MCPCatalog
+	if err := req.Get(&catalog, catalogName); err != nil {
+		return fmt.Errorf("failed to get catalog: %w", err)
+	}
+
+	if catalog.Annotations[v1.MCPCatalogSyncAnnotation] != "" {
+		delete(catalog.Annotations, v1.MCPCatalogSyncAnnotation)
+	} else {
+		if catalog.Annotations == nil {
+			catalog.Annotations = make(map[string]string)
+		}
+		catalog.Annotations[v1.MCPCatalogSyncAnnotation] = "true"
+	}
+
+	return req.Update(&catalog)
+}
+
+// Create creates a new catalog.
+func (h *MCPCatalogHandler) Create(req api.Context) error {
+	var manifest types.MCPCatalogManifest
+	if err := req.Read(&manifest); err != nil {
+		return fmt.Errorf("failed to read catalog manifest: %w", err)
+	}
+
+	// Validate the URLs
+	for _, urlStr := range manifest.SourceURLs {
+		u, err := url.Parse(urlStr)
+		if err != nil {
+			return fmt.Errorf("invalid URL: %w", err)
+		}
+
+		if u.Scheme != "https" {
+			return fmt.Errorf("only HTTPS URLs are supported")
+		}
+	}
```

Comment:
Do we want to allow the admin to do file paths as well? If you work around this API handler by using kubectl to directly modify the catalogs, you can just point them to directories and that works without any issues.